cargo/flatpak-cargo-generator.py

```diff
@@ -305,12 +316,22 @@ async def generate_sources(cargo_lock, git_tarballs=False):
         VENDORED_SOURCES: {'directory': f'{CARGO_CRATES}'},
     }
 
-    pkg_coros = [get_package_sources(p, cargo_lock, git_repos) for p in cargo_lock['package']]
+    if packages:
+        packages = packages.split(',')
+        logging.debug('Considering only packages: %s', packages)
+        all_packages = {p['name']: p for p in cargo_lock['package']}
```

Comment:
The `all_packages` dict feels out of place here, since it's only used within `packages_dependencies()` scope. Better pass the whole `cargo_lock` to `packages_dependencies()`, and there get whatever information is needed from it.